src/repository/views.py

```diff
@@ -274,7 +294,7 @@ def repository_search(request, search_term=None):
         )]
 
         preprints = list(set(list(preprint_search) + preprints_from_author))
-
+        preprints.sort(key=lambda x: x.date_published, reverse=True)
```

Comment:
Worth using `operator.attrgetter` here to squeeze some nanoseconds out of the sorting.